var/spack/repos/builtin/packages/py-googleapis-common-protos/package.py

```diff
@@ -26,14 +28,20 @@ class PyGoogleapisCommonProtos(PythonPackage):
         description="Enable support for gRPC Remote Procedure Call framework.",
     )
 
-    with when("+grpc"):
-        depends_on("py-grpcio@1.44:1", when="@1.57:", type="run")
-        depends_on("py-grpcio@1", when="@1.56.2:", type="run")
-        depends_on("py-grpcio@1:", type="run")
-
     depends_on("py-setuptools", type="build")
-    depends_on(
-        "py-protobuf@3.19.5:3.19,3.20.2:4.21.0,4.21.6:4", when="@1.58:", type=("build", "run")
-    )
-    depends_on("py-protobuf@3.12.0:3", when="@1.55", type=("build", "run"))
-    depends_on("py-protobuf@3.6.0:3", when="@:1.6", type=("build", "run"))
+
+    with default_args(type=("build", "run")):
+        # https://github.com/googleapis/python-api-common-protos/blob/main/setup.py
+        # May be able to rais max version to :5 in next release
+        depends_on("py-protobuf@3.19.5:4", when="@1.58:")
+        depends_on("py-protobuf@3.15.0:4", when="@1.56:1.57")
+        depends_on("py-protobuf@3.12.0:4", when="@1.55")
+        depends_on("py-protobuf@3.6.0:", when="@1.6.0:")
+
+    # Explicitly incompatibile versions per setup.py
+    conflicts("py-protobuf@3.20:3.21.1")
+    conflicts("py-protobuf@4.21.1:4.21.5")
```

Comment:
These two seem like they highlight troublesome versions upstream. 

The former may not be strict enough: https://github.com/protocolbuffers/protobuf/issues/10571. It seems like there were some hard to pin down incompatibilities in the 3.[19|20|21].[1|2|3] timeframe.

It sounds like the bug was present before but just didn't manifest: https://github.com/protocolbuffers/protobuf/issues/10116 
It may be safer to leave the `4.21.1:4.21.5`

Reply:
It also looks like the issue was present before that but they finally but the conflicts in the setup.py file in 1.57: https://github.com/googleapis/python-api-common-protos/issues/128

Reply:
I'd be inclined to leave these as-is due to the issues reported around those versions.